operator/config/manager/manager.yaml

```diff
@@ -43,7 +43,6 @@ spec:
           args:
             - --leader-elect
             - --health-probe-bind-address=:8081
-            - --metrics-bind-address=:8383
```

Comment:
I think there's an argument for keeping the port as-is considering the v1 chart spits out a ServiceMonitor that targets [:8383](https://github.com/pulumi/pulumi-kubernetes-operator/blob/d02ae39ed293efd2d779f3a10a530944ea054338/deploy/helm/pulumi-operator/templates/servicemonitor.yaml#L16) instead of the port name. It would be nice for the user's metrics to Just Work™ -- although if we're changing up the keys significantly then maybe there's no point.

In any case we should probably try to keep the metrics service as close as possible to what we have in v1, to at least keep selectors the same.

Maybe unrelated - it seems like the v1 operator has _two_ metrics ports, 8383 and 8686? Do you know why that is?

Reply:
PKO v1 was scaffolded with `operator-sdk` pre v1, which did not use `controller-runtime`/`kubebuilder` under the hood. It seems like the older version of the framework was modelled using `kube-state-metrics`  (where 2 ports were exposed, one for the operator, and another for metrics on the CRs itself, [ref](kube-state-metrics)). Kubebuilder uses a more simplified approach which exposes a globabl prometheus registry that we can add additional metrics to.

With the implementation of metrics being consolidated into 1 port, and industry standards moving to secure metrics by default, now seemed like a good time to make this change. WDYT?

Reply:
Yeah totally agree one port is the most reasonable. AFAICT the other port wasn't actually used in v1. I'm still wondering if it's OK to change the port, though. Existing service monitors would stop scraping metrics, right?